mypy/binder.py

```diff
@@ -15,7 +15,7 @@
 BindableExpression = Union[IndexExpr, MemberExpr, NameExpr]
 
 
-class Frame(Dict[Key, Type]):
+class Frame(Dict[Optional[Key], Type]):
```

Comment:
Does the base class really have to use `Optional[Key]` as its key type? Does `None` really occur as a valid key?